assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/stateListeners/usePriceTypeChangeListener.ts

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useRef, useState } from 'react';
+import { assocPath, pathOr } from 'ramda';
+
+import { AnyObject } from '@appServices/utilities/types';
+import { useDataState } from '../data';
+import { useUpdatePriceTypeForPrice } from '../utils';
+import { StateChangeListenerHook } from './types';
+
+const usePriceTypeChangeListener: StateChangeListenerHook = () => {
+	const { getData, prices, updatePrice } = useDataState();
+
+	const updatePriceTypeForPrice = useUpdatePriceTypeForPrice();
+
+	/**
+	 * To store the previous values of priceTypes for all prices
+	 * {
+	 *     [price.id]: price.priceType
+	 * }
+	 */
+	const priceTypeMapping = useRef<AnyObject<string>>({});
+
+	const [priceTypesStr, setPriceTypesStr] = useState('');
+	useEffect(() => {
+		// To avoid triggering the change on every render
+		// convert to JSON to only trigger when the value changes
+		const newPriceTypesStr = JSON.stringify(prices.map((price) => price.priceType));
+		setPriceTypesStr(newPriceTypesStr);
+	}, [prices]);
+	// Because of the deps, it will run only when priceTypes changes
+	useEffect(() => {
+		const newPriceTypeMapping = prices.reduce((acc, price) => {
+			return assocPath([price.id], price.priceType, acc);
+		}, {});
```

Comment:
can move contents of second `useEffect` into the first and wrap in a conditional that compares `priceTypesStr` against `newPriceTypesStr` before running the calculator